include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -366,12 +366,28 @@ struct __sycl_scan_by_segment_impl
     }
 };
 
+template <bool __is_inclusive, typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Range3,
+          typename _BinaryPredicate, typename _BinaryOperator, typename _T>
+void
+__parallel_scan_by_segment(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec, _Range1&& __keys,
+                           _Range2&& __values, _Range3&& __out_values, _BinaryPredicate __binary_pred,
+                           _BinaryOperator __binary_op, _T __init, _T __identity)
+{
+    __sycl_scan_by_segment_impl<__is_inclusive>()(
+        oneapi::dpl::__internal::__device_backend_tag{}, ::std::forward<_ExecutionPolicy>(__exec),
+        ::std::forward<_Range1>(__keys), ::std::forward<_Range2>(__values), ::std::forward<_Range3>(__out_values),
+        __binary_pred, __binary_op, __init, __identity);
+}
+} //namespace __par_backend_hetero
+
+namespace internal
```

Comment:
Is the `internal` or `__internal` namespace better here? This directory in general uses `internal` but oneDPL patterns tend to be placed in `__internal`. Perhaps this is better addressed when reworking the scan-by-segment algorithms to use reduce-then-scan. I expect to move these to the `pstl/hetero` directory similar to what was done for reduce-by-segment.

Reply:
good point.  Definitely the stronger convention in the repo is `__internal`, so lets do it while we are renaming things.

Reply:
> good point. Definitely the stronger convention in the repo is `__internal`, so lets do it while we are renaming things.

When the `by_segment` algorithms were added they did not follow all of the name uglification requirements of the rest of the oneDPL algorithms because they were not part of the algorithms that aligned with the C++17 algorithms.  Moving  to `__internal` to be consistent with the other algorithms while renaming is occurring is the correct thing to do in my opinion.